src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -173,7 +173,7 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
                            parent.IsKind(SyntaxKind.FixedStatement);
 
                 case SyntaxKind.ElseKeyword:
-                    return true;
+                    return token.Parent.IsKind(SyntaxKind.ElseClause);
```

Comment:
thnis is tested.  the main issue is that `#else $$` is not the beginning of a statement.